newly rewritten CCS+BCS+COL measures for HEDIS 2018

duncand opened the pull request:
This pull request includes 3 new CQL measures named with -2018 which rewrite the corresponding 3 measures for CCS+BCS+COL lacking the -2018 in their names.

That rewrite is a work in progress and this pull request has the work done so far, in order that review and feedback may be made on it.

This pull request should not actually be merged yet and the work will receive more iterations before it is truly "done".

duncand committed:
first draft of CCS measure 2018 version

duncand committed:
make ccs-logic-2018.cql work with FHIR 3 with no FHIRHelper

duncand committed:
fix off-by-one-year bugs in ccs-logic-2018.cql

duncand committed:
first draft of BCS measure 2018 version

duncand committed:
first draft of COL measure 2018 version

c-schuler commented on pages/cql/ccs-logic-2018.cql:
```diff
@@ -0,0 +1,163 @@
+	end
+
+define "Is Age 30 to 64 at End":
+	AgeInYearsAt(end of "Measurement Period") between 30 and 64
```
Where is the check to ensure the patient is 30 or older when the tests occurred as per step 2? The check you have in place ("Is Age 30 to 64 at End") only checks the age at the end of the measurement period.

duncand replied:
Thank you, I missed that.  It would go in the "Is Cervical Cytology Plus HPV Test In Last 5 Years" as an extra pair of "where" items.

c-schuler commented on pages/cql/ccs-logic-2018.cql:
```diff
@@ -0,0 +1,163 @@
+	exists(
+		"Dates of Cervical Cytology Tests" WhenCC
+			with "Dates of HPV Tests" WhenHPV
+				such that ((difference in days between (WhenCC as dateTime).value and (WhenHPV as dateTime).value) <= 4)
```
difference in years?

c-schuler replied:
Oh, I think that is the check for the same date of service, yes? In that case the dates should be equal, right?

c-schuler commented on pages/cql/ccs-logic-2018.cql:
```diff
@@ -0,0 +1,163 @@
+define "Is Cervical Cytology Test In Last 3 Years":
+	exists(
+		"Dates of Cervical Cytology Tests" WhenCC
+			where (WhenCC as dateTime).value during "Lookback Interval Two More Years"
```
Casting to dateTime here, but during operator is an Interval operation. See Included In operator in the spec for more details:
included in precision (left Interval<T>, right Interval<T>) Boolean

c-schuler replied:
Also, I believe if WhenCC is a Period and you cast it to dateTime (without specifying the start or end of the Period), the returned value would be an empty list.

c-schuler reviewed (changes requested):
Review for the CCS logic. Looks solid, but I believe a few tweaks are needed.
General Notes:
We either need to account for both Period and DateTime temporal types or restrict the qicore profiles to using only one temporal type.

The BCS and COL logic looks really good. You may need to incorporate suggested changes from CCS into the other libraries. Besides that, it is some damn fine looking CQL =)

duncand commented on pages/cql/ccs-logic-2018.cql:
```diff
@@ -0,0 +1,163 @@
+	end
+
+define "Is Age 30 to 64 at End":
+	AgeInYearsAt(end of "Measurement Period") between 30 and 64
```
The define "Is In Eligible Population" already checks that the patient is aged 24-64 which is the same as the step 1 requirement, so if the patient is in the eligible population they qualify for the step 1 test.  Only the step 2 test requires a tighter range of ages, 30-64.

brynrhodes reviewed (approved):
This is a great start, and I agree with Chris' comments on the review. Let's get those changes applied and merge this to the master so we can take next steps.

duncand committed:
in 2018 measures, properly handle instant/period duality

duncand committed:
add missing age 30 filter to 2018 CCS measure

duncand commented:
My last 2 commits addressed all of the issues raised by Chris' review comments.  All the measures were updated to properly handle the instant/period duality.  The CCS measure was also updated to add the missing age 30+ test.  Please review again, and merge if approved.

I request however that the dd_HEDIS_2018 branch NOT be deleted at the end of the merge process as I would like to commit more work using that branch rather than creating a new branch, work that addresses other outstanding gaps in the logic.

c-schuler commented:
Looks good Darren. Thank you.
